### PR TITLE
156 city resource generation

### LINKED ISSUE #156: Refactor City Resource Generation
- Make Cities generate resources over time instead of on click

### PR DESCRIPTION
**Pull request by Kuczek8**
Fixes #156 

Changes proposed in this pull request:
- Refactored City Class
- Gold generated on turn end
- Only generated if under your control (simple boolean for now)

### CODE REVIEW

File: Assets/Scripts/Strategy/GameResources/City.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+using System.Collections.Generic;
+using UnityEngine;
+using SwordAndBored.Strategy.TimeSystem.Subscribers;
+using SwordAndBored.Strategy.TimeSystem.TimeManager;
+
+namespace SwordAndBored.Strategy.GameResources
+{
+    public class City : MainThreadPostTimeStepSubscriber
+    {
+        public ResourceManager resourceManager { get; set; }
```
Comment (wynn-rj) [inline Assets/Scripts/Strategy/GameResources/City.cs]: Should be camel case

File: Assets/Scripts/Strategy/GameResources/City.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        public bool underPlayerControl = true;
+
+        void Awake()
+        {
+            underPlayerControl = true;
+        }
```
Comment (wynn-rj) [inline Assets/Scripts/Strategy/GameResources/City.cs]: Add a new line between methods

File: Assets/Scripts/Strategy/GameResources/City.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using SwordAndBored.Strategy.TimeSystem.Subscribers;
+using SwordAndBored.Strategy.TimeSystem.TimeManager;
```
Comment (wynn-rj) [inline Assets/Scripts/Strategy/GameResources/City.cs]: Remove unused usings

File: Assets/Scripts/Strategy/GameResources/City.cs [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        {
+            underPlayerControl = true;
+        }
+        protected override void MainThreadPostTimeStepUpdate()
+        {
+            if(underPlayerControl)
+            {
+                resourceManager.GoldAmount += 1;
+            }
+        }
```
Comment (wynn-rj) [inline Assets/Scripts/Strategy/GameResources/City.cs]: Should a city increase gold amount? Or should there be a GoldCity that extends this class that increases gold

**Review by wynn-rj — CHANGES_REQUESTED** [review]

File: Assets/Scripts/Strategy/GameResources/City.cs (hunk 1 above)
Comment (Kuczek8) [reply Assets/Scripts/Strategy/GameResources/City.cs]: Fixed in commit 0c427fe

File: Assets/Scripts/Strategy/GameResources/City.cs (hunk 2 above)
Comment (Kuczek8) [reply Assets/Scripts/Strategy/GameResources/City.cs]: Fixed in commit 0c427fe

File: Assets/Scripts/Strategy/GameResources/City.cs (hunk 3 above)
Comment (Kuczek8) [reply Assets/Scripts/Strategy/GameResources/City.cs]: Fixed in commit 0c427fe

File: Assets/Scripts/Strategy/GameResources/City.cs (hunk 4 above)
Comment (Kuczek8) [reply Assets/Scripts/Strategy/GameResources/City.cs]: Fixed in commit 0c427fe

**Review by cthomas68 — APPROVED** [review]

**Review by wynn-rj — APPROVED** [review]

**Review by subashch6 — APPROVED** [review]

**Pull request merged**